tools/ci-cdk/canary-lambda/src/s3_canary.rs

```diff
@@ -63,8 +63,14 @@ pub async fn s3_canary(client: s3::Client, s3_bucket_name: String) -> anyhow::Re
     let mut result = Ok(());
     match output.metadata() {
         Some(map) => {
-            if map.get("something") != Some(&"テスト テスト!".to_string()) {
-                result = Err(CanaryError("S3 metadata was incorrect".into()).into());
+            // Option::as_deref doesn't work here since the deref of &String is String
+            let value = map.get("something").map(|s| s.as_str()).unwrap_or("");
```

Comment:
I don't totally understand why the old code didn't work, can you explain?

Reply:
There's no change in the logic. I just rearranged things so that I could output the actual value received as part of the error.